apps/full-site-editing/full-site-editing-plugin/lib/feature-flags/feature-flags.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+class A8C_Full_Site_Editing_Feature_Flags {
+	private $flags = array();
+
+	public function __construct() {
+		$this->set_flags();
+	}
+
+	private function set_flags() {
+		$flags = '';
+
+		if ( isset( $_GET[ 'fse_flags' ] ) ) {
+			if ( empty( $_GET[ 'fse_flags' ] ) ) {
+				setcookie( 'fse_flags', '', time() - 3600 );
+				$this->flags = array();
+				return;
+			}
+
+			setcookie( 'fse_flags', $_GET[ 'fse_flags' ] );
+			$flags = $_GET[ 'fse_flags' ];
+		} else if( isset( $_COOKIE[ 'fse_flags' ] ) ) {
+			$flags = $_COOKIE[ 'fse_flags' ];
+		}
+
+		$flags = explode( ',', $flags );
+		foreach( $flags as $flag ) {
+			if ( 0 === strpos( $flag, '-' ) ) {
+				$this->flags[ substr( $flag, 1 ) ] = false;
+			} else {
+				$this->flags[ $flag ] = true;
+			}
+		}
+	}
+
+	public function get_flags() {
+		return $this->flags;
+	}
+
+	public function is_enabled( $flag ) {
+		if ( ! isset( $flag ) ) {
+			return false;
+		}
+		return in_array( $flag, $this->flags );
```

Comment:
This checks if the given flag exists as a value in the given array, but we need to check if the flag is a key of the array, and return its value (since that's what determines if the flag is enabled or not).

```suggestion
		return array_key_exists( $flag, $this->flags ) ? $this->flags[ $flag ] : false;
```

Reply:
Resolved in 011120a668d1d0887a6a013658f1c06b012a69d5 (in a different way).